src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1285,9 +1268,35 @@ patch(const xrt::module& module, const std::string& argnm, size_t index, const x
 }
 
 void
-patch(const xrt::module& module, const std::string& argnm, size_t index, uint64_t address)
+patch(const xrt::module& module, uint8_t *buf, size_t *sz, const std::vector< std::pair<std::string, uint64_t> > *args)
```

Comment:
```suggestion
patch(const xrt::module& module, uint8_t* buf, size_t* sz, const std::vector<std::pair<std::string, uint64_t>>& args)
```
Please document this function.   It's not obvious that with `buf==nullptr`, the function returns the size the caller should allocate for `buf`.   What is `args` representing?

Reply:
The args here representing a list of arguments that needs to be patched to the control code buffer. I have added comments to make it clear that how caller can discover the size for buffer allocation. I still need to keep args as a pointer (v.s. a reference) since when caller tries to discover the real size, it does not pass in a real arg list (nullptr would do).